pkgs/by-name/cd/cd-up/package.nix

```diff
@@ -0,0 +1,39 @@
+{
+  fetchFromGitHub,
+  gcc,
+  gnumake,
+  lib,
+  stdenv,
+}:
+stdenv.mkDerivation {
+  pname = "up-core";
+  name = "up";
+  version = "1.0.0";
+  src = fetchFromGitHub {
+    owner = "NewDawn0";
+    repo = "up";
+    rev = "1be487a84782bc62593326c35160fa782e2e7d7a";
+    hash = "sha256-Ll6gHuWh2s7ke9Vqlw0H3tdKHIuvIC13TGHHKllEQ54=";
+  };
+  installPhase = ''
+    mkdir -p $out/bin $out/lib
+    cp up-core $out/bin
+    echo "#!/usr/bin/env bash" > $out/lib/SOURCE_ME.sh
+    $out/bin/up-core --init >> $out/lib/SOURCE_ME.sh
+    chmod +x $out/lib/SOURCE_ME.sh
+  '';
+  shellHook = ''
+    source $out/lib/SOURCE_ME.sh
+  '';
```

Comment:
What is this for?

Reply:
Packages like up and dirstack have a backend written in C or Rust, this binary cannot change directories. My approach to solving this problem is creating a file which when sourced provides the up and ds functions which executes the cd command the backend provides them with, thereby changing directories in the user's shell. If there is a better approach, I'll gladly change it

Reply:
I'm not sure I follow. In any case this sounds like something that users should set in their shellInit